inc/shortcodes/attributions/class-attachments.php

```diff
@@ -83,14 +82,33 @@ function ( $excluded_shortcodes ) {
 	}
 
 	/**
-	 * Gets all attachments currently in the database, returns as an array of
+	 * Returns the array of attachments set in the instance variable
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array|null
+	 */
+	function getBookMedia() {
+		// Cheap cache
+		static $media = null;
+		if ( null !== $media ) {
+			return $media;
+		} else {
+			$media = $this->book_media;
+		}
+
+		return $media;
+	}
```

Comment:
Can you clarify this approach?  I feel it would be confusing in this context:

```php
$one = new Attachments();
$one->setBookMedia();
$results = $one->getBookMedia();

// 1000 lines of code later

$two = new Attachments();
$results = $two->getBookMedia();

// $results will contain whatever $one set/get
```

The kind of caching I was thinking of was more like:

```php
$a = new Attachments();
$r1 = $a->getBookMedia(); // This queries the database
$r2 = $a->getBookMedia(); // Cached in memory, no database,
$r3 = $a->getBookMedia(); // Cached in memory, no database,
// ...
```

However, since we now have both  `setBookMedia()` and `getBookMedia()`, and set does all the work, then I feel the method could simply be:

```
function getBookMedia() {
   return $this->book_media;
}
```

The developer would be repsonsible for (re)setting as needed, using the same instance of the class as needed, etc.

Thoughts?

Reply:
CC: @alex-418